### ISSUE TITLE
[1.12] Essentials /tell is broken (uses vanilla /tell instead)

### ISSUE THREAD

**Issue by Sikatsu** [EssentialsX/Essentials#1285]
When someone messages me with /tell: http://prntscr.com/fhe7zp

When I message someone with /tell: http://prntscr.com/fhe86c

Also nicknames and such don't work, so I guess it uses vanilla /tell.

**Comment by Sikatsu** [comment]
@SupaHam Hey, are you able to look into this issue perhaps?

**Comment by SupaHam** [comment]
Sorry, I'm a quite busy these days. I hope to get back to these issues soon.

**Comment by Sikatsu** [comment]
Appreciated.

**Comment by bigsmileling** [comment]
Inform us if its fixed please, everyone is using it for know vanished staff

**Comment by SupaHam** [comment]
Disable the minecraft permission to use `/tell` which should be `minecraft.command.tell`.

SpigotMC, if not Mojang, has made a breaking change with command aliases. So you'll find `/msg` works as expected, but `/tell` is an alias of `/msg` so it is a lower priority and so Minecraft is allowed to override it. Still thinking of a fix for the problem.

**Comment by Sikatsu** [comment]
I removed the permission, but still hoping it can be fixed somehow as it seems nearly everyone is used to using /tell.

**Comment by SupaHam** [comment]
After much work, I have discovered the change that causes this behaviour.

SpigotMC proceeded to remove some deprecated parts of the Bukkit API. One of those things was Bukkit's built-in equivalent of vanilla commands. Without going into much detail (although commits are below) The pre 1.12 code would make plugin commands a priority over vanilla commands. However, after the removal of this API, and some logic code Minecraft commands are prioritised, but not minecraft command aliases (e.g. `/w`, `/msg`, etc.)

I've informed the SpigotMC team via IRC of this issue and wait to hear back regarding a fix. Until then, there is not much I can do to solve this issue in a proper manner. I don't believe this was their intention and hope for the same results you do. :)

The API commit removing `VanillaCommand` class: [link](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/commits/a1cc4b769caf690b2f075f833c83e76d6431f9a7)

The CB commit removing usage of `VanillaCommand` class: [link](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/5195487ec6c01a80e3a2fca8c4acfc6b9524fcf5#src/main/java/org/bukkit/craftbukkit/command/VanillaCommandWrapper.java)
